hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -211,10 +236,53 @@ public Response getOpenKeyInfo(
         break;
       }
     }
+    // Populate the keysSummary map
+    createKeysSummaryForOpenKey(keysSummary);
+
+    openKeyInsightInfo.setKeysSummary(keysSummary);
+
     openKeyInsightInfo.setLastKey(lastKey);
     return Response.ok(openKeyInsightInfo).build();
   }
 
+  /**
+   * Creates a keys summary for open keys and updates the provided
+   * keysSummary map. Calculates the total number of open keys, replicated
+   * data size, and unreplicated data size.
+   *
+   * @param keysSummary A map to store the keys summary information.
+   */
+  private void createKeysSummaryForOpenKey(
+      Map<String, Object> keysSummary) {
+    Long replicatedSizeOpenKey = getValueFromId(globalStatsDao.findById(
+        OmTableInsightTask.getReplicatedSizeKeyFromTable(OPEN_KEY_TABLE)));
+    Long replicatedSizeOpenFile = getValueFromId(globalStatsDao.findById(
+        OmTableInsightTask.getReplicatedSizeKeyFromTable(OPEN_FILE_TABLE)));
+    Long unreplicatedSizeOpenKey = getValueFromId(globalStatsDao.findById(
+        OmTableInsightTask.getUnReplicatedSizeKeyFromTable(OPEN_KEY_TABLE)));
+    Long unreplicatedSizeOpenFile = getValueFromId(globalStatsDao.findById(
+        OmTableInsightTask.getUnReplicatedSizeKeyFromTable(OPEN_FILE_TABLE)));
+    Long openKeyCountForKeyTable = getValueFromId(globalStatsDao.findById(
+        OmTableInsightTask.getTableCountKeyFromTable(OPEN_KEY_TABLE)));
+    Long openKeyCountForFileTable = getValueFromId(globalStatsDao.findById(
+        OmTableInsightTask.getTableCountKeyFromTable(OPEN_FILE_TABLE)));
```

Comment:
Do we need consider multi-part upload in-progress case? May be another JIRA to report open key summary

Reply:
I believe a separate jira would be better!